src/components/Tutorial/__tests__/Tutorial.ts

```diff
@@ -0,0 +1,296 @@
+import { getByText, screen } from '@testing-library/dom'
+import userEvent from '@testing-library/user-event'
+import { act } from 'react'
+import {
+  HOME_TOKEN,
+  TUTORIAL2_STEP_CHOOSE,
+  TUTORIAL2_STEP_CONTEXT1,
+  TUTORIAL2_STEP_CONTEXT1_PARENT,
+  TUTORIAL2_STEP_CONTEXT1_SUBTHOUGHT,
+  TUTORIAL2_STEP_CONTEXT2_PARENT,
+  TUTORIAL2_STEP_CONTEXT_VIEW_EXAMPLES,
+  TUTORIAL2_STEP_CONTEXT_VIEW_OPEN,
+  TUTORIAL2_STEP_CONTEXT_VIEW_SELECT,
+  TUTORIAL2_STEP_START,
+  TUTORIAL_STEP_AUTOEXPAND,
+  TUTORIAL_STEP_AUTOEXPAND_EXPAND,
+  TUTORIAL_STEP_FIRSTTHOUGHT,
+  TUTORIAL_STEP_FIRSTTHOUGHT_ENTER,
+  TUTORIAL_STEP_SECONDTHOUGHT,
+  TUTORIAL_STEP_SECONDTHOUGHT_ENTER,
+  TUTORIAL_STEP_START,
+  TUTORIAL_STEP_SUBTHOUGHT,
+  TUTORIAL_STEP_SUBTHOUGHT_ENTER,
+  TUTORIAL_STEP_SUCCESS,
+} from '../../../constants'
+import exportContext from '../../../selectors/exportContext'
+import store from '../../../stores/app'
+import createTestApp, { cleanupTestApp, cleanupTestEventHandlers } from '../../../test-helpers/createTestApp'
+
+// as per https://testing-library.com/docs/user-event/options/#advancetimers
+// we should avoid using { delay: null }, and use jest.advanceTimersByTime instead
+const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime })
+
+/** Get last created thought in the document, used mostly after `user.keyboard('{Enter}')`. */
+const lastThought = () => Array.from(document.querySelectorAll('[contenteditable="true"]')).at(-1)!
```

Comment:
Let's use `[aria-label="thought"] [contenteditable]` or `[data-editable="true"]` to be a bit more specific. There happen to be no other contenteditable's on the screen, but it's a bit safer to not assume that.

Even better would be to specifically select the empty thought.

Reply:
> Even better would be to specifically select the empty thought.

I considered this during my previous review, but didn't mention it since the last thought and the empty thought were coincidentally always the same.

It would be nicer to explicitly select the thought to be edited (the empty thought), and easier to update in the future if we need to edit another thought during testing.

Reply:
So, I arrived at 
```tsx
/** Get last created empty thought in the document. Gets the last thought that is not empty. Mostly used after `user.keyboard('{Enter}')` to get the new thought. */
const lastThought = () =>
  Array.from(document.querySelectorAll('[data-editable="true"]'))
    .filter(it => !it.textContent)
    .at(-1)!
```

I think it works. Perhaps, a more refined solution could be out there, but it works :D
I'd prefer to modify this thing and move out to `test-helpers` at some point if we hit the wall, but not now.